### PR TITLE
MAINT: NEP29 Apr 2024, zenodo updates

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

- Updates NEP29 standards for Apr 5, 2024.
- Updates zenodo -> author order, institutions, keywords

# Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

via GitHub Actions main.yml

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
Note that test failures caused in part by https://github.com/pysat/pysat/issues/1196

**Comment by jklenzing** [comment]
Adding the "new_test" flag to dmsp_ssusi fixes the memory issue. The remaining issue is in pysat. Could comment out the clean_warn flags with an issue to fix for the next release.

**Comment by jklenzing** [comment]
Pinging @aburrell for comments on the path forward

**Comment by aburrell** [comment]
> Adding the "new_test" flag to dmsp_ssusi fixes the memory issue. The remaining issue is in pysat. Could comment out the clean_warn flags with an issue to fix for the next release.

This seems like the best path forward.

**Comment by jklenzing** [comment]
So.... tests for python 3.10 and 3.12 work on ubuntu, and tests for 3.11 work on windows, and everything else is failing with some sort of netCDF4 / HDF error. This particular combination is failing consistently.

**Comment by jklenzing** [comment]
NOTE: test_w_pad now fails for maven_sep, presumably due to new xarray releases. Documented in #222.

**Comment by aburrell** [comment]
Looks like you may need to update coveralls to use Coveralls Actions or follow the steps I took at OCBPy.

**Comment by aburrell** [comment]
Since the coveralls coverage-reporter is failing regardless of what I do, I have asked support for their help.

**Review by aburrell — APPROVED** [review]
Recommend we open an issue for the Mac Coveralls failure and move on with our development.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -11,12 +11,12 @@ jobs:
...
         os: ["ubuntu-latest", "macos-latest", "windows-latest"]
-        python-version: ["3.10", "3.11"]
+        python-version: ["3.11", "3.12"]
         numpy_ver: ["latest"]
         test_config: ["latest"]
         include:
-          - python-version: "3.9"
-            numpy_ver: "1.21"
+          - python-version: "3.10"
+            numpy_ver: "1.23"
```
Comment (jklenzing) [inline .github/workflows/main.yml:19]:
```suggestion
            numpy_ver: "1.24"
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,6 +36,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
   * Use standard clean routine for C/NOFS VEFI mag data
   * Added version cap for sphinx_rtd_theme
   * Include standard tests for ICON IVM-B
+  * Update NEP29 standards for Apr 2024
```
Comment (jklenzing) [inline CHANGELOG.md:39]:
```suggestion
  * Update NEP29 standards for Jun 2024
```

**Comment by jklenzing** [comment]
Added #232 to cover the open issue. Also tweaked the minimum numpy version for the June 2024 update.

**Pull request merged**